lib/application/pages/home/collections/collections_list_view.dart

```diff
@@ -0,0 +1,235 @@
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:layoutr/common_layout.dart';
+import 'package:memo/application/constants/colors.dart' as colors;
+import 'package:memo/application/constants/dimensions.dart' as dimens;
+import 'package:memo/application/constants/strings.dart' as strings;
+import 'package:memo/application/theme/theme_controller.dart';
+import 'package:memo/application/view-models/home/collections_vm.dart';
+import 'package:memo/application/widgets/animatable_progress.dart';
+import 'package:memo/application/widgets/theme/themed_text_tag.dart';
+import 'package:memo/core/faults/errors/inconsistent_state_error.dart';
+
+class CollectionsListView extends HookWidget {
+  const CollectionsListView() : super();
+
+  @override
+  Widget build(BuildContext context) {
+    final state = useProvider(collectionsVM.state);
+
+    if (state is LoadingCollectionsState) {
+      return const Center(child: CircularProgressIndicator());
+    }
+
+    final loadedState = state as LoadedCollectionsState;
+    final items = loadedState.collectionItems;
+
+    return ListView.builder(
+      itemCount: items.length,
+      itemBuilder: (context, index) {
+        // Builds the respective widget based on the item's `CollectionItemMetadata` subtype
+        final item = items[index];
```

Comment:
I liked the way how you built `loadedState.collectionItems` adding the headers in the same array as the items so the list can build them in the correct order